### DISCUSSION TITLE
[Discussion] Do we need frontrun prevention for the fraud proof?

### DISCUSSION THREAD

**Discussion by ChihChengLiang** [thehubbleproject/RedditHubble#444] [General]
This is an issue to document our discussion and decision on the topic.

TLDR:  front-running helps fraud-proof showing up faster, the benefit is greater than harm.

@kilic Can add a commit-open method to prevent fraud-proof frontrunning

@barrywhitehat  Right so the argument is that if we have commit reveal only watcher nodes can compete to include fraud-proof transactions. If we don’t have commit reveal then watcher nodes + front runners must compete. So if front runners are included then they will bid up transactions as they get broadcast.
So with commit reveal watcher nodes have to be able to bid high enough to get a transaction included. Without it, they just have to broadcast some transaction and other nodes will bid it up.

Also, it makes the fraud-proof a one-step process instead of two which makes censorship attacks harder.

@vaibhavchellani  frontrunning might reduce incentives for running watcher nodes

@barrywhitehat I think this is a concern. But I would argue. That 

1. Frauds will be so rare that the incentive to run node just for fraud is pretty low. 
2. There will likely be so many nodes that the actual returns are even lower. 
3. There will already be people highly competitive for claiming frauds so even adding commit reveal will not save this incentive.

**Comment by ChihChengLiang** [comment]
Reopen or referencing this issue when we have something new to discuss
